core/dbt/task/run.py

```diff
@@ -645,7 +650,6 @@ def safe_run_hooks(
             return RunStatus.Success
 
         status = RunStatus.Success
-        failed = False
```

Comment:
Thanks for making this logic more concise. 

If I understand it correctly, we want the behavior to be: if a node fails, we want to skip the rest?

I think the direction of `status` and `failed` is duplicate is spot on, but I am wondering should we get rid of status and rename failed to something like `skip_rest_due_to_failure` instead? In the current structure I have to do some guess and validation in order to understand what the code is trying to do.

Reply:
That is correct, if one _hook_ fails, the rest are expected to be skipped. 

I can clean up the code a bit more, I was trying to avoid too much refactoring in this PR as it's generally best practice to separate feature/bug work from refactors. I actually walked back some refactoring work I did in this regard.

---

To the keen eyed observer, [5e9d3d6](https://github.com/dbt-labs/dbt-core/pull/10936/commits/5e9d3d6bf5f7ad139d24e3e21421f7837550284f) foreshadowed some work that I actually dropped before pushing my branch:
> If it isn't apparent, I'm trying to reduce the need for tracking variables
to make it easier extract the execution code to a separate private function
to make it easier to see what is happening.